### PR TITLE
fix: [PL-23642]: OAS non-complaint components for 120 ng-manager

### PR DESCRIPTION
**Pull request by VikasMaddukuriHarness**
You can use the following comments to re-trigger PR Checks

- Compile: `trigger compile`
- runAeriformCheck: `trigger AeriformCheck`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`

---
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/32637)

### CODE REVIEW

**Comment by VikasMaddukuriHarness** [comment]
trigger messagecheck

**Comment by VikasMaddukuriHarness** [comment]
trigger ti2

File: 960-ng-core-beans/src/main/java/io/harness/ng/core/dto/AccountDTO.java [hunk 1]
```diff
@@ -38,14 +38,27 @@
...
-  @VariableExpression(skipVariableExpression = true) AuthenticationMechanism authenticationMechanism;
-  @VariableExpression(skipVariableExpression = true) ServiceAccountConfig serviceAccountConfig;
-  @VariableExpression(skipVariableExpression = true) boolean isNextGenEnabled;
+  @Schema(description = "Identifier of the account.")
+  @EntityIdentifier(allowBlank = false)
+  @VariableExpression(skipVariableExpression = true)
+  String identifier;
+  @Schema(description = "Name of the account.") @NGEntityName String name;
+  @Schema(description = "Name of the company name.") String companyName;
+  @Schema(description = "Name of the cluster associated with this account.")
```
Comment (rashminandasahoo) [inline 960-ng-core-beans/src/main/java/io/harness/ng/core/dto/AccountDTO.java:47]: Name of the cluster associated with this Account.

File: 960-ng-core-beans/src/main/java/io/harness/ng/core/dto/AccountDTO.java [hunk 2]
```diff
@@ -38,14 +38,27 @@
...
+  @VariableExpression(skipVariableExpression = true)
+  String cluster;
+  @Schema(description = "Specifies the default experience of the account.")
```
Comment (rashminandasahoo) [inline 960-ng-core-beans/src/main/java/io/harness/ng/core/dto/AccountDTO.java:50]: Default experience of the Account.

File: 960-ng-core-beans/src/main/java/io/harness/ng/core/dto/AccountDTO.java [hunk 3]
```diff
@@ -38,14 +38,27 @@
...
+  @VariableExpression(skipVariableExpression = true)
+  DefaultExperience defaultExperience;
+  @Schema(description = "Authentication mechanism associated with the account.")
+  @VariableExpression(skipVariableExpression = true)
+  AuthenticationMechanism authenticationMechanism;
+  @Schema(description = "Configuration of the service account associated with the account.")
```
Comment (rashminandasahoo) [inline 960-ng-core-beans/src/main/java/io/harness/ng/core/dto/AccountDTO.java:56]: Service Account configuration associated with this Account.

File: 960-ng-core-beans/src/main/java/io/harness/ng/core/dto/AccountDTO.java [hunk 4]
```diff
@@ -38,14 +38,27 @@
...
+  @VariableExpression(skipVariableExpression = true)
+  ServiceAccountConfig serviceAccountConfig;
+  @Schema(description = "Specifies whether or not Nextgen is enabled to the account.")
```
Comment (rashminandasahoo) [inline 960-ng-core-beans/src/main/java/io/harness/ng/core/dto/AccountDTO.java:59]: Specifies if NextGen is enabled for this Account.

File: 960-ng-core-beans/src/main/java/io/harness/ng/core/dto/UserGroupDTO.java [hunk 5]
```diff
@@ -36,23 +37,28 @@
 @JsonInclude(NON_NULL)
 @Schema(name = "UserGroup", description = "This is the view of the UserGroup entity defined in Harness")
```
Comment (rashminandasahoo) [inline 960-ng-core-beans/src/main/java/io/harness/ng/core/dto/UserGroupDTO.java:38]: User Group details defined in Harness.

File: 960-ng-core-beans/src/main/java/io/harness/ng/core/environment/dto/EnvironmentRequestDTO.java [hunk 6]
```diff
@@ -34,12 +35,14 @@
 @JsonIgnoreProperties(ignoreUnknown = true)
 @Schema(name = "EnvironmentRequest", description = "This is the EnvironmentRequest entity defined in Harness")
```
Comment (rashminandasahoo) [inline 960-ng-core-beans/src/main/java/io/harness/ng/core/environment/dto/EnvironmentRequestDTO.java:36]: Do not use DB nomenclature like "EnvironmentRequest". Please edit accordingly throughout.

File: 960-ng-core-beans/src/main/java/io/harness/ng/core/invites/dto/CreateInviteDTO.java [hunk 7]
```diff
@@ -32,8 +32,12 @@
...
-  @ApiModelProperty(required = true) InviteType inviteType;
-  List<String> userGroups;
+  @Schema(description = "List of users to be invited.")
+  @ApiModelProperty(required = true)
+  @NotEmpty
+  @Size(max = 100)
+  List<String> users;
+  @Schema(description = "Role bindings to be associated with the invited users.") List<RoleBinding> roleBindings;
+  @Schema(description = "Specifies the invite type.") @ApiModelProperty(required = true) InviteType inviteType;
+  @Schema(description = "List of the userGroups in the invite.") List<String> userGroups;
```
Comment (rashminandasahoo) [inline 960-ng-core-beans/src/main/java/io/harness/ng/core/invites/dto/CreateInviteDTO.java:42]: "user groups"

File: 960-ng-core-beans/src/main/java/io/harness/ng/core/service/dto/ServiceRequestDTO.java [hunk 8]
```diff
@@ -30,11 +31,11 @@
 @JsonInclude(NON_NULL)
 @Schema(name = "ServiceRequest", description = "This is the ServiceRequest entity defined in Harness")
```
Comment (rashminandasahoo) [inline 960-ng-core-beans/src/main/java/io/harness/ng/core/service/dto/ServiceRequestDTO.java:32]: Service Request details defined in Harness.

**Review by rashminandasahoo — APPROVED** [review]
:lgtm:

> Reviewed 21 of 33 files at r1, 12 of 12 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32637#-:-N1cM7d8DSaalZUxuwPu:bnfp4nl)* status: 33 file(s) to review. 1 of 2 approvals obtained.

**Review by rashminandasahoo — APPROVED** [review]
> Reviewed 6 of 6 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32637)* status: 33 file(s) to review. 1 of 2 approvals obtained.

**Review by nishantsaini20 — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32637)* status: 33 file(s) to review. 1 change requests. 1 of 2 approvals obtained. 

___
*[`120-ng-manager/src/main/java/io/harness/ng/core/smtp/resources/SmtpNgResource.java` line 94 at r3](https://reviewable.io/reviews/harness/harness-core/32637#-N2V8T5e2XD1568imrA9:-N2V8T5e2XD1568imrAA:btnb04z) ([raw file](https://github.com/harness/harness-core/blob/deb1ace44e36cf1ed38318c7300355a0afcaa037/120-ng-manager/src/main/java/io/harness/ng/core/smtp/resources/SmtpNgResource.java#L94)):*
> ```Java
>   public ResponseDTO<NgSmtpDTO>
>   save(@Valid @NotNull NgSmtpDTO variable,
>       @NotNull @QueryParam(NGCommonEntityConstants.ACCOUNT_KEY) @AccountIdentifier String accountIdentifier)
> ```

@AccountIdentifier is for access control checks. Why removed at many places?

**Review by VikasMaddukuriHarness — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32637)* status: 33 file(s) to review. 1 change requests. 1. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`120-ng-manager/src/main/java/io/harness/ng/core/smtp/resources/SmtpNgResource.java` line 94 at r3](https://reviewable.io/reviews/harness/harness-core/32637#-N2V8T5e2XD1568imrA9:-N2WIqDW5bMViWqtjsCf:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/deb1ace44e36cf1ed38318c7300355a0afcaa037/120-ng-manager/src/main/java/io/harness/ng/core/smtp/resources/SmtpNgResource.java#L94)):*
<details><summary><i>Previously, nishantsaini20 (Nishant Saini) wrote…</i></summary><blockquote>

@AccountIdentifier is for access control checks. Why removed at many places?
</blockquote></details>

Done.

**Review by nishantsaini20 — CHANGES_REQUESTED** [review]
> Reviewed 4 of 5 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32637)* status: 33 file(s) to review. 1 change requests. 0 of 2 approvals obtained. 

___
*[`120-ng-manager/src/main/java/io/harness/ng/core/remote/NGSecretResourceV2.java` line 537 at r4](https://reviewable.io/reviews/harness/harness-core/32637#-N2WLVwcCG3yRgJuUM2F:-N2WLVwcCG3yRgJuUM2G:blyn122) ([raw file](https://github.com/harness/harness-core/blob/1b44a801c959697387619d2a529a6be780f55a76/120-ng-manager/src/main/java/io/harness/ng/core/remote/NGSecretResourceV2.java#L537)):*
> ```Java
>   public ResponseDTO<DecryptableEntity>
>   decryptEncryptedDetails(@Body DecryptableEntityWithEncryptionConsumers decryptableEntityWithEncryptionConsumers,
>       @Parameter(description = ACCOUNT_PARAM_MESSAGE, required = true) @AccountIdentifier @QueryParam(
> ```

Is @AccountIdentifier added intentionally here?
___
*[`120-ng-manager/src/main/java/io/harness/ng/core/remote/NGSecretResourceV2.java` line 560 at r4](https://reviewable.io/reviews/harness/harness-core/32637#-N2WLblm72qgAi9kKiHJ:-N2WLblm72qgAi9kKiHK:bha174) ([raw file](https://github.com/harness/harness-core/blob/1b44a801c959697387619d2a529a6be780f55a76/120-ng-manager/src/main/java/io/harness/ng/core/remote/NGSecretResourceV2.java#L560)):*
> ```Java
>   public ResponseDTO<List<EncryptedDataDetail>>
>   getEncryptionDetails(@NotNull NGAccessWithEncryptionConsumer ngAccessWithEncryptionConsumer,
>       @Parameter(description = ACCOUNT_PARAM_MESSAGE, required = true) @NotBlank @QueryParam(
> ```

@AccountIdentifier is removed?

**Comment by VikasMaddukuriHarness** [comment]
trigger ti0

**Review by nishantsaini20 — APPROVED** [review]
> Reviewed 2 of 2 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32637)* status: 33 file(s) to review. 1 of 2 approvals obtained.

File: 960-ng-core-beans/src/main/java/io/harness/ng/core/dto/SmtpConfigDTO.java [hunk 9]
```diff
@@ -33,11 +33,26 @@
...
+  @NotNull
+  private int port;
+  @Schema(description = "From address of the email that needs to be send.")
+  @JsonProperty("fromAddress")
+  private String fromAddress;
+  @Schema(description = "Specify whether or not to use SSL certificate.")
+  @JsonProperty("useSSL")
+  private boolean useSSL;
+  @Schema(description = "Specify whether or not to use TLS.") @JsonProperty("startTLS") private boolean startTLS;
+  @Schema(description = "UserName credential to authenticate with SMTP server.")
```
Comment (meenakshi-harness) [inline 960-ng-core-beans/src/main/java/io/harness/ng/core/dto/SmtpConfigDTO.java:52]: why capital N?

**Review by rashminandasahoo — APPROVED** [review]
> Reviewed 4 of 5 files at r4, 2 of 2 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32637)* status: 27 file(s) to review.

File: 960-ng-core-beans/src/main/java/io/harness/ng/core/dto/NgSmtpDTO.java [hunk 10]
```diff
@@ -26,8 +27,13 @@
 @Builder
 @Schema(name = "NgSmtp", description = "This is the view of the NgSmtp entity defined in Harness")
 public class NgSmtpDTO {
-  private String uuid;
-  @NotEmpty String accountId;
-  @NotNull @NotBlank @Trimmed(message = "The name must not have trailing spaces.") @NGEntityName private String name;
+  @Schema(description = "Identifier of the SMTP config.") private String uuid;
```
Comment (meenakshi-harness) [inline 960-ng-core-beans/src/main/java/io/harness/ng/core/dto/NgSmtpDTO.java:30]: Is this(uuid) field shown  in oas spac?

Comment (VikasMaddukuriHarness) [reply 960-ng-core-beans/src/main/java/io/harness/ng/core/dto/NgSmtpDTO.java:30]: Yes. This is the Identifier of the SMTP config. This is used in terms in update smtp call

File: 120-ng-manager/src/main/java/io/harness/ng/core/remote/TokenResource.java [hunk 11]
```diff
@@ -108,8 +109,9 @@ public class TokenResource {
         ApiResponse(responseCode = "default", description = "Returns created Token details")
       })
   public ResponseDTO<String>
-  createToken(@Valid TokenDTO tokenDTO,
-      @NotNull @QueryParam(NGCommonEntityConstants.ACCOUNT_KEY) @AccountIdentifier String accountIdentifier) {
+  createToken(@Parameter(description = ACCOUNT_PARAM_MESSAGE, required = true) @NotBlank @QueryParam(
```
Comment (meenakshi-harness) [inline 120-ng-manager/src/main/java/io/harness/ng/core/remote/TokenResource.java:112]: why removed notnull and changed it with notblank?

File: 120-ng-manager/src/main/java/io/harness/ng/core/remote/ApiKeyResource.java [hunk 12]
```diff
@@ -105,8 +107,9 @@ public class ApiKeyResource {
         ApiResponse(responseCode = "default", description = "Returns the created API key")
       })
   public ResponseDTO<ApiKeyDTO>
-  createApiKey(
-      @Valid ApiKeyDTO apiKeyDTO, @NotNull @QueryParam(ACCOUNT_KEY) @AccountIdentifier String accountIdentifier) {
+  createApiKey(@Parameter(description = ACCOUNT_PARAM_MESSAGE, required = true) @NotBlank @QueryParam(
```
Comment (meenakshi-harness) [inline 120-ng-manager/src/main/java/io/harness/ng/core/remote/ApiKeyResource.java:110]: why notnull changed to notblank?

File: 120-ng-manager/src/main/java/io/harness/ng/core/remote/NGSecretResourceV2.java [hunk 13]
```diff
@@ -558,7 +559,8 @@ private void validateRequestPayload(SecretRequestWrapper dto) {
   @InternalApi
   public ResponseDTO<List<EncryptedDataDetail>>
   getEncryptionDetails(@NotNull NGAccessWithEncryptionConsumer ngAccessWithEncryptionConsumer,
-      @NotNull @QueryParam(NGCommonEntityConstants.ACCOUNT_KEY) @AccountIdentifier String accountIdentifier) {
+      @Parameter(description = ACCOUNT_PARAM_MESSAGE, required = true) @NotBlank @AccountIdentifier @QueryParam(
```
Comment (meenakshi-harness) [inline 120-ng-manager/src/main/java/io/harness/ng/core/remote/NGSecretResourceV2.java:562]: same

**Review by VikasMaddukuriHarness — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32637)* status: 31 file(s) to review. 3. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`120-ng-manager/src/main/java/io/harness/ng/core/remote/ApiKeyResource.java` line 110 at r5](https://reviewable.io/reviews/harness/harness-core/32637#-N0lDNQ7h87lH-mewcjt-r5-110:-N2jk4Xz2sxey5hEC2cs:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/c68a9192697a88b54058962c32179353d430e74c/120-ng-manager/src/main/java/io/harness/ng/core/remote/ApiKeyResource.java#L110)):*
<details><summary><i>Previously, meenakshi-harness (Meenakshi Raikwar) wrote…</i></summary><blockquote>

why notnull changed to notblank?
</blockquote></details>

Done.
___
*[`120-ng-manager/src/main/java/io/harness/ng/core/remote/NGSecretResourceV2.java` line 562 at r5](https://reviewable.io/reviews/harness/harness-core/32637#-N0lDNQ7h87lH-mewcju-r5-562:-N2jk6Yp7KhvHfCNiFUu:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/c68a9192697a88b54058962c32179353d430e74c/120-ng-manager/src/main/java/io/harness/ng/core/remote/NGSecretResourceV2.java#L562)):*
<details><summary><i>Previously, meenakshi-harness (Meenakshi Raikwar) wrote…</i></summary><blockquote>

same
</blockquote></details>

Done.

**Review by meenakshi-harness — APPROVED** [review]
> Reviewed 15 of 33 files at r1, 8 of 12 files at r2, 1 of 5 files at r4, 1 of 2 files at r5, 5 of 5 files at r6, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32637)* status: 8 file(s) to review. 1 of 2 approvals obtained.

**Review by gupta-utkarsh — COMMENTED** [review]
> Reviewed 4 of 33 files at r1, 2 of 12 files at r2, 4 of 5 files at r6, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32637)* status: 4 file(s) to review. 1 of 2 approvals obtained.

**Review by nishantsaini20 — APPROVED** [review]
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32637)* status: 4 file(s) to review.

**Review by meenakshi-harness — APPROVED** [review]
> Reviewed 3 of 6 files at r3.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32637)* status: 1 file(s) to review.

**Review by nishantsaini20 — APPROVED** [review]
> Reviewed 15 of 33 files at r1, 8 of 12 files at r2, 3 of 6 files at r3, 5 of 5 files at r6.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32637)* status: 3 file(s) to review.

**Review by manish-manchanda — APPROVED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32637#-:-N2kAKHl2_JecBQhXBeQ:bnfp4nl)* status: 3 file(s) to review.

**Review by nishantsaini20 — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32637)* status: 3 file(s) to review.

**Review by diptiman-adak — APPROVED** [review]
:lgtm:

> Reviewed 15 of 33 files at r1, 8 of 12 files at r2, 1 of 5 files at r4, 1 of 2 files at r5, 5 of 5 files at r6, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32637#-:-N2l7wr4B1l5_F_2xg-t:bnfp4nl)* status: 3 file(s) to review.

**Review by diptiman-adak — APPROVED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32637#-:-N2l93Dp4w8I6z8bUW5y:bnfp4nl)* status: 3 file(s) to review.

**Review by manish-manchanda — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32637)* status: 3 file(s) to review.

**Review by ankit-harness — APPROVED** [review]
:lgtm:

> Reviewed 15 of 33 files at r1, 8 of 12 files at r2, 1 of 5 files at r4, 1 of 2 files at r5, 5 of 5 files at r6, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32637#-:-N2oa8Ma55m9BDkofnnm:bnfp4nl)* status: 3 file(s) to review.

**Review by reetika04 — APPROVED** [review]
> Reviewed 15 of 33 files at r1, 8 of 12 files at r2, 3 of 6 files at r3, 1 of 5 files at r4, 1 of 2 files at r5, 5 of 5 files at r6, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32637)* status: 3 file(s) to review.

**Review by reetika04 — APPROVED** [review]
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32637)* status: 3 file(s) to review.

**Review by reetika04 — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32637)* status: 3 file(s) to review.

**Review by ankit-harness — APPROVED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32637#-:-N2ocvKRE6RszquQ1M5o:bnfp4nl)* status: 3 file(s) to review.

**Review by nishantsaini20 — APPROVED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32637#-:-N2odCdiA_Ak0ebWOhk7:bnfp4nl)* status: 3 file(s) to review.

**Pull request merged**